tools/devtool

```diff
@@ -375,6 +388,102 @@ run_devctr() {
         "$DEVCTR_IMAGE" "${ctr_args[@]}"
 }
 
+# Given a list of packages it adds them to {arch-based}-requirements_freeze.txt
+#
+# Args:
+#   $1  path to the {arch-based}-requirements-freeze file.
+#
+perform_update() {
+    declare requirements_freeze_file_location="$1"
+    shift
+    for package_name in "$@"; do
+        echo "$package_name" >>$requirements_freeze_file_location
+    done
+    echo "" >> $requirements_freeze_file_location
+}
+
+# Looks for any package which are present in requirements.txt file but
+# not in {arch-based}-requirements-freeze.txt file. Prompts the user if they want
+# to install and add freeze their version in {arch-based}-requirements_freeze.txt
+# 
+# Args:
+#   $1  path to the {arch-based}-requirements-freeze file.
+# 
+update_packages() {
+    declare requirements_freeze_file_location="$1"  
+    declare parsed_package_location=$(mktemp)
+    say "Checking for newly added requirements..."
+    while IFS= read -r line; do
+        echo "${line%%=*}" >>$parsed_package_location
+    done <"$requirements_freeze_file_location"
+    unset IFS
+
+    diff=($(grep -v -f $parsed_package_location $FC_DEVCTR_DIR/requirements/requirements.txt))    
+    rm $parsed_package_location
+
+    if [ "${#diff[@]}" -ne 0 ]; then
```

Comment:
Tested this locally and it does not handle the removed packages. We should also add support for this use case

Reply:
That's a good call-out. Removing a package is little tricky. 

For addition logic,  we check packages that are present in ```requirements.txt``` but not in ``requirements-freeze.txt```.

We might have to do reverse for deletion logic, check for packages that are in  ```requirements-freeze.txt``` but deleted from  ``requirements-freeze.txt``.  But that won't be accurate,  ``requirements-freeze.txt``` always has more packages than ```requirements.txt``` . 

We can just maintain a single file ```requirements-freeze.txt```(per arch) and add/remove packages directly to this file which solves the problem of addition and deletion. 

@gc-plp @alindima let me know what you guys think about this.

Reply:
I think this is a very good idea. This would also simplify the code

Reply:
resolving this comment as we changed the approach and moved to use Poetry.